Added documentation and test coverage.

palimondo opened the pull request:
compare_perf_test.py is now covered with unit tests and public methods are documented in the implementation.

Minor refactoring  to better conform to Python conventions:
* classes declared in new style
* proper private method prefix of single underscore
* replacing map with list comprehension where it was clearer

Unit test are executed as part of validation-test.

.gitignore was modified to ignore `.coverage` and `htmlcov` artifacts generated by the `coverage.py` package

palimondo committed:
Added documentation and test coverage.

compare_perf_test.py is now covered with unit tests and public methods are documented in the implementation.

Minor refactoring  to better conform to Python conventions:
* classes declared in new style
* proper private method prefix of single underscore
* replacing map with list comprehension where it was clearer

Unit test are executed as part of validation-test.

.gitignore was modified to ignore .coverage and htmlcov artifacts generated by the coverage.py package

palimondo commented:
@shahmishal @gottesmm Please review.

palimondo commented:
W8, I need to fix lint violations…

CodaFi commented:
@swift-ci please python lint

CodaFi commented:
@swift-ci please python lint

palimondo committed:
Improved coverage with more tests: parse_args

Coverage at 66% according to coveragy.py

shahmishal commented:
@swift-ci please smoke test

xwu commented on benchmark/scripts/compare_perf_tests.py:
```diff
@@ -21,8 +21,22 @@
+    from Swift Benchmark Suite as reported by test driver (Benchmark_O,
+    Benchmark_Onone, Benchmark_Ounchecked or Benchmark_Driver).
+
+    It depends on the log format emmited by the test driver in the form:
```
Typo: the correct spelling is "emitted"

xwu commented on benchmark/scripts/compare_perf_tests.py:
```diff
@@ -21,8 +21,22 @@
+    It depends on the log format emmited by the test driver in the form:
+    #,TEST,SAMPLES,MIN(μs),MAX(μs),MEAN(μs),SD(μs),MEDIAN(μs),MAX_RSS(B)
+
+    The last column, MAX_RSS, is emmited only for runs instrumented by the
```
Same typo

xwu commented on benchmark/scripts/compare_perf_tests.py:
```diff
@@ -21,8 +21,22 @@
 
-class PerformanceTestResult:
+class PerformanceTestResult(object):
+    """PerformanceTestResult holds results from executing individual benchmark
```
Missing articles: "from executing _an_ individual benchmark from _the_ Swift Benchmark Suite..."

xwu reviewed (commented):
Spelling and grammar nits.

palimondo committed:
Improved coverage: ReportFormatter

Coverage at 87% according to coveragy.py

Also fixed spelling errors in documentation.

palimondo commented:
@shahmishal Sorry for pushing another change while the smoke test was running, but I'd hate to open extra PR just to fix the grammar nits @xwu brought up. I had a couple more test ready, so I slipped those in, too. Please re-🚬

palimondo commented:
@gottesmm Is the unit test integration with validation-test done OK here? How about the placement of test themselves? You didn't comment on that aspect on #9999. Can you review this, please?

shahmishal commented:
@swift-ci please smoke test

palimondo commented:
@shahmishal @gottesmm @dabrahams @eeckstein @atrick Can somebody please review and merge this? Otherwise I can't help myself and will keep tinkering with it...

palimondo commented:
@CodaFi @natecook1000 Could you? ^^

dabrahams commented:
@swift-ci Please test and merge

dabrahams commented:
@swift-ci Please test and merge

dabrahams commented:
@palimondo This really should be run through the full validation suite, but if that passes, it will merge.

Thanks for your hard work on this!

palimondo committed:
Consistent --output; Improved coverage: main()

Coverage at 99% according to coverage.py

* `compare_perf_tests.py` now always outputs the same format to stdout as is written to `--output` file
* Added integration test for the main() function
* Added tests for console output (and suppressed it leaking during testing)
* Fixed file name in test’s file header

palimondo commented:
OMG, Sorry @dabrahams haven't seen your response (GitHub didn't update until I refreshed browser) and I pushed a final commit. Could you please re-run the CI?

palimondo committed:
Fixed false statement in documentation.

palimondo committed:
One more typo fix.

palimondo commented:
Ah, I'm so clumsy... fixing everything in 1000 steps. I would have force pushed, but it seems to mess with CI, requiring it to run twice, so I added two more commits instead. 🤦‍♂️

natecook1000 commented:
@swift-ci Please test and merge

palimondo commented:
@natecook1000 You can test, but not merge?

palimondo commented:
@practicalswift Any idea why this didn't merge after passing the test?

palimondo commented:
@shahmishal Thanks for the merge!
